SwiftRuntimeLibrary/SwiftMarshal/DynamicLib.cs

```diff
@@ -86,6 +95,16 @@ public DynamicLib (string fileName, DLOpenMode flags)
 
 		}
 
+		IEnumerable<string> LibraryPaths ()
+		{
+			var path = Environment.GetEnvironmentVariable ("DYLD_LIBRARY_PATH");
+			if (path == null)
+				yield break;
+			foreach (var subPart in path.Split (':')) {
+				yield return subPart;
+			}
```

Comment:
I think something like this should work:

```suggestion
			if (path == null)
				return Array.Empty<string> ();
			return path.Split (':');
```